src/classic/clvm_tools/stages/stage_2/module.rs

```diff
@@ -385,26 +397,110 @@ fn compile_mod_stage_1(
                         &mut namespace,
                         &mut functions,
                         &mut constants,
+                        &mut delayed_constants,
                         &mut macros,
                         run_program.clone()
                     )?;
                 }
 
-                let uncompiled_main = alist[alist.len() - 1];
-                m! {
-                    main_list <-
-                        enlist(
+                // For each delayed constant, drain it into the
+                // main constant pool.
+                let mut result_collection = CollectionResult {
+                    functions,
+                    constants,
+                    macros
+                };
+
+                let main_name_vec = MAIN_NAME.as_bytes().to_vec();
+
+                // Process delayed constants until we either can't advance or
+                // they're all done.
+
+                loop {
+                    if delayed_constants.is_empty() {
+                        break;
+                    }
+
+                    let mut processed = false;
+                    // copy so we can modify delayed_constants.
+                    let delayed_constant_defs: Vec<(Vec<u8>, NodePtr)> =
+                        delayed_constants.iter().map(|(k,v)| (k.clone(), *v)).collect();
+
+                    for (name, delayed_body) in delayed_constant_defs.iter() {
```

Comment:
I believe `.iter()` is redundant here, isn't it?

Reply:
So far clippy hasn't bugged me.  I'll check.

Reply:
no because that'd consume the collection, and each pass only takes some of the items.